### PR TITLE
Update the GUI of our application

### PR DESCRIPTION
**Pull request by wujy28**
# Update the GUI of our application
Improved the GUI of our application to more closely match our intended use and ui draft
---
# Key changes

## Created Record Panel to display a patient's record
1. Reformatted the layout of the GUI to match our UI design
2. Added a Record Panel to view the patient's record

## Added GUI interaction to view selected patient's record
1. Added functionality whereby if you click and select a patient in the patient list, you can view their complete record
in the record panel.

Unfortunately I did not have time to properly style the GUI theme to match our branding but I'll definitely do it the next few iterations. Also, I might have introduced like a bug or two when I changed some files (the default values or something like that) to get the code to compile so we'll have to fix them 🥲

### CODE REVIEW

File: src/main/resources/view/MainWindow.fxml [hunk 1]
```diff
@@ -8,11 +8,12 @@
...
 <?import javafx.scene.control.SplitPane?>
 <?import javafx.scene.image.Image?>
+<?import javafx.scene.layout.AnchorPane?>
 <?import javafx.scene.layout.StackPane?>
 <?import javafx.scene.layout.VBox?>
+<?import javafx.stage.Stage?>
 
-<fx:root type="javafx.stage.Stage" xmlns="http://javafx.com/javafx/17" xmlns:fx="http://javafx.com/fxml/1"
-         title="Address App" minWidth="450" minHeight="600" onCloseRequest="#handleExit">
+<fx:root minHeight="600" minWidth="450" onCloseRequest="#handleExit" title="Advanced&amp;Efficient" type="javafx.stage.Stage" xmlns="http://javafx.com/javafx/17" xmlns:fx="http://javafx.com/fxml/1">
```
Comment (thienmy0) [inline src/main/resources/view/MainWindow.fxml:16]: Hi! May I ask if the title here supposed to be "Advanced&Efficient" or is it "Advanced&amp;Efficient" on purpose?

File: src/test/java/seedu/address/testutil/RecordBuilder.java [hunk 2]
```diff
@@ -28,9 +28,9 @@ public RecordBuilder() {
      */
     public RecordBuilder(PatientBuilder patient) {
         this.patient = patient.build();
-        this.initialObservations = Record.DEFAULT_INITIAL_OBSERVATIONS;
-        this.diagnosis = Record.DEFAULT_DIAGNOSIS;
-        this.treatmentPlan = Record.DEFAULT_TREATMENT_PLAN;
+        this.initialObservations = Record.getDefaultInitialObservations();
+        this.diagnosis = Record.getDefaultDiagnosis();
+        this.treatmentPlan = Record.getDefaultTreatmentPlan();
```
Comment (thienmy0) [inline src/test/java/seedu/address/testutil/RecordBuilder.java:33]: I like how you fixed the compilation error here! 😋

File: src/test/java/seedu/address/testutil/TypicalPatients.java [hunk 3]
```diff
@@ -2,12 +2,8 @@
 
 import static seedu.address.logic.commands.CommandTestUtil.VALID_ADDRESS_AMY;
 import static seedu.address.logic.commands.CommandTestUtil.VALID_ADDRESS_BOB;
-import static seedu.address.logic.commands.CommandTestUtil.VALID_BIRTHDAY_AMY;
-import static seedu.address.logic.commands.CommandTestUtil.VALID_BIRTHDAY_BOB;
 import static seedu.address.logic.commands.CommandTestUtil.VALID_EMAIL_AMY;
 import static seedu.address.logic.commands.CommandTestUtil.VALID_EMAIL_BOB;
-import static seedu.address.logic.commands.CommandTestUtil.VALID_GENDER_AMY;
-import static seedu.address.logic.commands.CommandTestUtil.VALID_GENDER_BOB;
```
Comment (AaronJT1) [inline src/test/java/seedu/address/testutil/TypicalPatients.java:10]: Would it be possible to also add the birthday and gender attribute for Bob and Amy below?

File: src/main/resources/view/RecordCard.fxml
Comment (AaronJT1) [inline src/main/resources/view/RecordCard.fxml:1]: I like how you created aa new RecordCard specifically for outr project. 👍

**Review by AaronJT1 — APPROVED** [review]
Overall the UI code seems to be good. Nice job researching on how to create the UI using javaFX!! Comments given are trivial so changes could be done in next iteration as well. But if so, do add it in github issues tagged to the relevant milestone. Job well done 👍

**Pull request merged**